### PR TITLE
intital debug for saving transformations

### PR DESCRIPTION
**Pull request by danieldaugbjergusd**
# Overview

**Type of Change:** Bug Fix

**Summary:** Move transformation methods into initial dictionary in TrainingProgram where they are accessed for preprocessing and saving.

## Model Updates

N/A

### Data Models

N/A

### Object-Oriented Models

Training methods and save method now access transformation methods from dictionary

## End-to-End Testing Instructions

Run simulator

### CODE REVIEW

File: src/training_program.py [hunk 1]
```diff
@@ -62,6 +62,28 @@ def __init__(self, dataframe, class_column , num_classes):
                 self.class_string_dictionary[class_values] = class_to_idx[class_values]
                 self.class_set.add(class_to_idx[class_values])
 
+        # Create transformation method dictionary
```
Comment (tmckeown311) [inline src/training_program.py:65]: Having the transformations defined all in one place is very convenient for future changes and testing purposes

File: src/training_program.py [hunk 2]
```diff
@@ -470,24 +466,28 @@ def save_models(self, model_filenames = None, height_filename = None,
             caud_file = model_filenames["caud"]
             caud_filename = os.path.join("src/models", caud_file)
             torch.save(self.caud_model.state_dict(), caud_filename)
+            self.save_transformation(self.transformations["caud"], 0)
```
Comment (tmckeown311) [inline src/training_program.py:469]: saving transformations only when models are updated is a good change to prevent discrepancies between transformations used when training the used model

**Review by tmckeown311 — APPROVED** [review]
Changes look good and should help with preventing transformation changes when model accuracy does not increase.
